packages/ui-components/src/components/DateTimePicker/DateTimePicker.component.tsx

```diff
@@ -0,0 +1,651 @@
+  invalid?: boolean
+  /** The label of the DateTimePicker input element. */
+  label?: string
+  /** Localization string or object. Can be used to set starting day of the week, e.g. Mondays instead of Sundays. More on localization: https://flatpickr.js.org/localization/ */
+  locale?: LocaleKey | Partial<CustomLocale> | null
+  /** The maximum / latest date a user can select (inclusive). */
+  maxDate?: DateOption | null
+  /** The minimum / earliest date a user can select (inclusive). */
+  minDate?: DateOption | null
+  /**  The step for the minute input. Only has an effect when a time picker is enabled via `enableTime`.  */
+  minuteIncrement?: number
+  /** The mode of the Datepicker. */
+  mode?: "single" | "multiple" | "range"
+  /** Whether to show a dropdown to select the current month. Default is "static". If `showMonths` is set to be greater than 1, it will always be displayed as static. Arrows to scroll through the months as well as through years will still be displayed and working. */
+  monthSelectorType?: "static" | "dropdown"
+  /** The name of the DateTimePicker input element */
+  name?: string
+  /** Set to `true` to not display a calendar at all. To create a time picker, set `enableTime` to true, too. */
+  noCalendar?: boolean
+  /** A handler to be executed when the DateTimePicker input element looses focus. */
+  // eslint-disable-next-line no-unused-vars
+  onBlur?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected date(s), date range or time changes */
+  // eslint-disable-next-line no-unused-vars
+  onChange?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker value is reset by clicking the clear icon. The onChnage handler will be fired in this event too, onClear is more specific. */
+  // eslint-disable-next-line no-unused-vars
+  onClear?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker calendar closes */
+  // eslint-disable-next-line no-unused-vars
+  onClose?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker input element receives focus. */
+  // eslint-disable-next-line no-unused-vars
+  onFocus?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected month changes */
+  // eslint-disable-next-line no-unused-vars
+  onMonthChange?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker calendar opens */
+  // eslint-disable-next-line no-unused-vars
+  onOpen?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the DateTimePicker component is ready */
+  // eslint-disable-next-line no-unused-vars
+  onReady?: (date?: Date[], dateStr?: string) => void
+  /** A handler to be executed when the selected year changes */
+  // eslint-disable-next-line no-unused-vars
+  onYearChange?: (date?: Date[], dateStr?: string) => void
```

Comment:
**(Blocking)**

Please make sure these method inputs etc. are backwards compatible too.

Reply:
In this case, the functions are custom implementations and are unrelated to HTMLElement. While they share the same name, their functionality differs. Grzegorz implemented them correctly, as evidenced by how they are being used in the code.

Reply:
I made it match "func" as closesly as possible in recent update

Reply:
Thanks, @andypf. The current implementation was [my suggestion](https://github.com/cloudoperators/juno/pull/640#discussion_r1860374989). Just when I was thinking about it again, I wanted to make sure that it's backward compatible.

In [this comment](https://github.com/cloudoperators/juno/pull/640#discussion_r1863145788) I was just sharing examples where the input was more general. In this case, it could be e.g. `(event?: any) => void`.

I'm just trying to make sure my suggestion was correct for what was allowed in the past.

And `(date?: Date[], dateStr?: string) => void` would be an improvement. Similar to backward compatibility consideration `[here](https://github.com/cloudoperators/juno/pull/640#discussion_r1863406393)`.